src/lib/sendRequest.js

```diff
@@ -26,32 +29,70 @@ export default (opt) => (dispatch) => {
   if (opt.withCredentials) {
     req.withCredentials()
   }
+}
 
-  req.end((err, res) => {
-    if (!res && !err) {
-      err = new Error(`Connection failed: ${debug}`)
+export const handleRequest = (params, { onTail, onStandard }) => {
```

Comment:
The only issue I can see with this is that we have to repackage that params object every time we pass it down which happens quite a lot here, do you think that's better?  Or is there a way with destructuring to pull out variables and keep the wrapping object? I searched around but I could find any.